### ISSUE TITLE
Error while installing with brew (macOS 10.14)

### ISSUE THREAD

**Issue by bmrs** [breakfastquay/rubberband#69]
Hi,

Having quite an old computer, I'm stuck with macOS 10.14. Until this day, there was no problem to install `rubberband`, via `homebrew`, as a dependency of `ffmpeg`. But since the last `brew upgrade`, the build fails:

```
[1/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_faster_SilentAudioCurve.cpp.o -MF librubberband_objlib.a.p/src_faster_SilentAudioCurve.cpp.o.d -o librubberband_objlib.a.p/src_faster_SilentAudioCurve.cpp.o -c ../src/faster/SilentAudioCurve.cpp
[2/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_faster_HighFrequencyAudioCurve.cpp.o -MF librubberband_objlib.a.p/src_faster_HighFrequencyAudioCurve.cpp.o.d -o librubberband_objlib.a.p/src_faster_HighFrequencyAudioCurve.cpp.o -c ../src/faster/HighFrequencyAudioCurve.cpp
[3/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_common_Profiler.cpp.o -MF librubberband_objlib.a.p/src_common_Profiler.cpp.o.d -o librubberband_objlib.a.p/src_common_Profiler.cpp.o -c ../src/common/Profiler.cpp
[4/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_common_Log.cpp.o -MF librubberband_objlib.a.p/src_common_Log.cpp.o.d -o librubberband_objlib.a.p/src_common_Log.cpp.o -c ../src/common/Log.cpp
[5/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_rubberband-c.cpp.o -MF librubberband_objlib.a.p/src_rubberband-c.cpp.o.d -o librubberband_objlib.a.p/src_rubberband-c.cpp.o -c ../src/rubberband-c.cpp
[6/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_faster_AudioCurveCalculator.cpp.o -MF librubberband_objlib.a.p/src_faster_AudioCurveCalculator.cpp.o.d -o librubberband_objlib.a.p/src_faster_AudioCurveCalculator.cpp.o -c ../src/faster/AudioCurveCalculator.cpp
[7/30] clang++ -Ilibrubberband-jni.dylib.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband-jni.dylib.p/src_jni_RubberBandStretcherJNI.cpp.o -MF librubberband-jni.dylib.p/src_jni_RubberBandStretcherJNI.cpp.o.d -o librubberband-jni.dylib.p/src_jni_RubberBandStretcherJNI.cpp.o -c ../src/jni/RubberBandStretcherJNI.cpp
FAILED: librubberband-jni.dylib.p/src_jni_RubberBandStretcherJNI.cpp.o 
clang++ -Ilibrubberband-jni.dylib.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband-jni.dylib.p/src_jni_RubberBandStretcherJNI.cpp.o -MF librubberband-jni.dylib.p/src_jni_RubberBandStretcherJNI.cpp.o.d -o librubberband-jni.dylib.p/src_jni_RubberBandStretcherJNI.cpp.o -c ../src/jni/RubberBandStretcherJNI.cpp
../src/jni/RubberBandStretcherJNI.cpp:26:10: fatal error: 'system/Allocators.h' file not found
#include "system/Allocators.h"
         ^~~~~~~~~~~~~~~~~~~~~
../src/jni/RubberBandStretcherJNI.cpp:26:10: note: did not find header 'Allocators.h' in framework 'system' (loaded from '/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks')
1 error generated.
[8/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_common_Thread.cpp.o -MF librubberband_objlib.a.p/src_common_Thread.cpp.o.d -o librubberband_objlib.a.p/src_common_Thread.cpp.o -c ../src/common/Thread.cpp
[9/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_common_Allocators.cpp.o -MF librubberband_objlib.a.p/src_common_Allocators.cpp.o.d -o librubberband_objlib.a.p/src_common_Allocators.cpp.o -c ../src/common/Allocators.cpp
[10/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_faster_PercussiveAudioCurve.cpp.o -MF librubberband_objlib.a.p/src_faster_PercussiveAudioCurve.cpp.o.d -o librubberband_objlib.a.p/src_faster_PercussiveAudioCurve.cpp.o -c ../src/faster/PercussiveAudioCurve.cpp
[11/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_common_StretchCalculator.cpp.o -MF librubberband_objlib.a.p/src_common_StretchCalculator.cpp.o.d -o librubberband_objlib.a.p/src_common_StretchCalculator.cpp.o -c ../src/common/StretchCalculator.cpp
[12/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_faster_CompoundAudioCurve.cpp.o -MF librubberband_objlib.a.p/src_faster_CompoundAudioCurve.cpp.o.d -o librubberband_objlib.a.p/src_faster_CompoundAudioCurve.cpp.o -c ../src/faster/CompoundAudioCurve.cpp
[13/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_common_Resampler.cpp.o -MF librubberband_objlib.a.p/src_common_Resampler.cpp.o.d -o librubberband_objlib.a.p/src_common_Resampler.cpp.o -c ../src/common/Resampler.cpp
[14/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_common_sysutils.cpp.o -MF librubberband_objlib.a.p/src_common_sysutils.cpp.o.d -o librubberband_objlib.a.p/src_common_sysutils.cpp.o -c ../src/common/sysutils.cpp
[15/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_faster_StretcherChannelData.cpp.o -MF librubberband_objlib.a.p/src_faster_StretcherChannelData.cpp.o.d -o librubberband_objlib.a.p/src_faster_StretcherChannelData.cpp.o -c ../src/faster/StretcherChannelData.cpp
[16/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_faster_StretcherProcess.cpp.o -MF librubberband_objlib.a.p/src_faster_StretcherProcess.cpp.o.d -o librubberband_objlib.a.p/src_faster_StretcherProcess.cpp.o -c ../src/faster/StretcherProcess.cpp
[17/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_common_FFT.cpp.o -MF librubberband_objlib.a.p/src_common_FFT.cpp.o.d -o librubberband_objlib.a.p/src_common_FFT.cpp.o -c ../src/common/FFT.cpp
[18/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_RubberBandStretcher.cpp.o -MF librubberband_objlib.a.p/src_RubberBandStretcher.cpp.o.d -o librubberband_objlib.a.p/src_RubberBandStretcher.cpp.o -c ../src/RubberBandStretcher.cpp
[19/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_faster_R2Stretcher.cpp.o -MF librubberband_objlib.a.p/src_faster_R2Stretcher.cpp.o.d -o librubberband_objlib.a.p/src_faster_R2Stretcher.cpp.o -c ../src/faster/R2Stretcher.cpp
[20/30] clang++ -Ilibrubberband_objlib.a.p -I. -I.. -I../rubberband -I../src -I/usr/local/Cellar/libsamplerate/0.1.9_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -Wpedantic -std=c++11 -O3 -arch x86_64 -mmacosx-version-min=10.13 -DHAVE_VDSP -DHAVE_LIBSAMPLERATE -DLACK_SINCOS -DNO_THREAD_CHECKS -DNO_TIMING -DNDEBUG -DUSE_PTHREADS -DMALLOC_IS_ALIGNED -MD -MQ librubberband_objlib.a.p/src_finer_R3Stretcher.cpp.o -MF librubberband_objlib.a.p/src_finer_R3Stretcher.cpp.o.d -o librubberband_objlib.a.p/src_finer_R3Stretcher.cpp.o -c ../src/finer/R3Stretcher.cpp
ninja: build stopped: subcommand failed.

HOMEBREW_VERSION: 3.6.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: b8b195cc64a29595797651720ebb2ea09affb682
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 7c509e94d988c6b54c13f4af3db47c5ad4c504a2
Core tap last commit: 62 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_NO_BOOTSNAP: set
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: dodeca-core 64-bit westmere
Clang: 11.0.0 build 1100
Git: 2.21.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
macOS: 10.14.6-x86_64
CLT: 11.3.1.0.1.1576735732
Xcode: 11.3.1

HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j12
CMAKE_PREFIX_PATH: /usr/local
CMAKE_INCLUDE_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/libxml2:/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
PKG_CONFIG_PATH: /usr/local/opt/libogg/lib/pkgconfig:/usr/local/opt/flac/lib/pkgconfig:/usr/local/opt/libvorbis/lib/pkgconfig:/usr/local/opt/opus/lib/pkgconfig:/usr/local/opt/libsndfile/lib/pkgconfig:/usr/local/opt/libsamplerate/lib/pkgconfig
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.14
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/local/opt/meson/bin:/usr/local/opt/ninja/bin:/usr/local/opt/pkg-config/bin:/usr/local/opt/flac/bin:/usr/local/opt/libsndfile/bin:/usr/local/opt/libsamplerate/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

I'm thinking that the 7th step `did not find header 'Allocators.h' in framework 'system' (loaded from '/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/System/Library/Frameworks')` is the cause of the problem: the folder does not exist, so I've tried to reinstall Xcode CommandLineTools (v11.3.1), both from terminal and by downloading directly the package from the [Apple Developper website](https://developer.apple.com/download/all/?q=Command%2011.3), but the folder with the required `Frameworks` is not installed where it is supposed to be.

And so the built cannot terminate, and I cannot use `ffmpeg`anymore… Thanks for any help!

**Comment by cannam** [comment]
Fixed in the repo now. Totally my fault; the fix is simply to change `system/Allocators.h` to `common/Allocators.h` in line 26 of `RubberBandStretcherJNI.cpp`.

(The error message that you saw is deceptive - having failed to find an include file called `system/Allocators.h` the compiler is jumping to the conclusion that it was intended to be in a framework called `system`. But that isn't the case; it was always just a file elsewhere within the Rubber Band repo, and the problem is simply that it has moved. And that nobody had exercised the JNI build since it moved! I've updated the build scripts as well so that the JNI library is now built during my normal development run.)
